### PR TITLE
fix iframe proxing (top level window uses AMD module loader) (fix #127)

### LINKED ISSUE #127: 'almond' AMD module loader crashes jsParsingTools module inside Iframe without src
'almond' AMD module loader - https://github.com/jrburke/almond

Page for reproduce without 'almond':
```javascript
<html>
	<head>
		<!--<script src="https://raw.githubusercontent.com/jrburke/almond/latest/almond.js"></script>-->
		<script>
			var define;
			define = function () {};
	    	define.amd = {}
		</script>
	</head>
	<body>
		<h1>Test</h1>
		<script>
			var iframe = document.createElement('iframe');
			iframe.id = "iframe";
			document.body.appendChild(iframe);
		</script>
	</body>
</html>
```

After opening page in playground I get the script error:
'Uncaught TypeError: Cannot read property 'parse' of undefined'

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @inikulin @churkin 
The test name was checked by @MargaritaLoseva

### CODE REVIEW

File: test/client/fixtures/sandbox/iframe-test.js [hunk 1]
```diff
@@ -43,6 +43,24 @@ test('event should not raise if a cross-domain iframe is appended', function ()
     $iframe.remove();
 });
 
+asyncTest('the AMD module loader damages proxing an iframe without src', function () {
```
Comment (inikulin) [inline test/client/fixtures/sandbox/iframe-test.js]: It should be in `regression` section and test title should inlclude github issue number (in format `(GH-{num})`)

**Comment by inikulin** [comment]
It has so much insertions/deletions because of reformatting, right?

\r-

**Comment by miherlosev** [comment]
>>It has so much insertions/deletions because of reformatting, right?
Yes

**Comment by miherlosev** [comment]
Commit contains fix for unstable test (test/client/fixtures/sandbox/event/focus-blur-change-test.js)
FPR

File: test/client/fixtures/sandbox/event/focus-blur-change-test.js [hunk 2]
```diff
@@ -770,12 +770,22 @@ if (window.HTMLInputElement.prototype.setSelectionRange) {
...
+                    attemptCount++;
+                    var isInputFocused = document.activeElement === input2;
+
+                    if (focusRaised === needFocus && isInputFocused === needFocus ) {
+                        expect(0);
+                        window.clearInterval(intervalId);
+                        startNext();
+                    }
+                    else if (attemptCount > maxAttemptCount)
+                        ok(false, 'The check for calling focus failed');
```
Comment (inikulin) [inline test/client/fixtures/sandbox/event/focus-blur-change-test.js]: @VasilyStrelyaev Check spelling pls

Comment (VasilyStrelyaev) [inline test/client/fixtures/sandbox/event/focus-blur-change-test.js]: The check actually succeeds - it's the result that is unexpected, so maybe:
'The element's focus state doesn't match the expected one.'

**Comment by miherlosev** [comment]
FRP
/cc @LavrovArtem

**Comment by inikulin** [comment]
Maybe we can wrap this code with attempts into some kind of function in `test-utils` or smthg like this, so it can be reused later?

**Comment by inikulin** [comment]
Let's discuss it on Monday with @AlexanderMoskovkin and @churkin

**Comment by miherlosev** [comment]
ping @AlexanderMoskovkin , @churkin

**Comment by AlexanderMoskovkin** [comment]
You can implement this in the `before-test.js` file or create the `test-utils.js` file and add it to the `CLIENT_TESTS_SETTINGS` in the `Gulpfile.js`.
I suppose we can use this function in the following way:
```js
//the util function
window.checkWithInterval = function (predicate, done, interval, timeout) {
...
}

//test
asyncTest('test', function () {
    function checkSomething () {
        return a === b && c === d;
    }
    function done (success) {
        if(!success)
            ok(false, 'our check are failed');
    
        start();
    }
    
    window.checkWithInterval(checkSomethink, done, 100, 1000);
});
```

As an alternative we can implement some build-in function in the `qunit-harness`. Something like:
```js
//test
asyncTest('test', function () {
    window.QUnitGlobals.assert(function (ok, notOk, equal, notEqual) {
         ok(someValue);
    }, interval, timeout);
});
```
In this case qunit-harness can automatically make assertions during timeout and finish the test if assertions successed or timeout expired

**Comment by inikulin** [comment]
@AlexanderMoskovkin What I don't like in current implementation is what we don't use assertions, so it's not clear what checked in the file. The good implementation will be to add `attempts` parameter to the test and restart test on qunit level if it fails (some kind of quarantine mode but for qunit). I suppose we should modify qunit-harness

**Comment by AlexanderMoskovkin** [comment]
Created an issue for the `qunit-harness` repo: https://github.com/AlexanderMoskovkin/qunit-harness/issues/6

**Comment by inikulin** [comment]
@miherlosev Let's keep things as is right now, but create separate issue for switching tests you modified to the AlexanderMoskovkin/qunit-harness#6 mechanism once it will be ready.

**Comment by miherlosev** [comment]
Commit contains only bug fix.
Separate issue for unstable tests - https://github.com/superroma/testcafe-hammerhead/issues/136

**Comment by miherlosev** [comment]
FPR

**Comment by miherlosev** [comment]
ping @inikulin @churkin

**Comment by churkin** [comment]
Restore the original `parsing-tools.js` code format please

**Comment by churkin** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**